sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -188,45 +174,44 @@ private static void WriteMemory(MemoryStream stream, ReadOnlyMemory<byte> memory
         /// ]]></code>
         /// </example>
         /// <remarks> See error response code and error response message for more detail. </remarks>
-        public virtual Response<UploadLogsResult> Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
```

Comment:
It looks like the Response will just be from whichever request happened to complete last. This seems unintuitive. I wonder if we should even return a Response at all.

/cc @tg-msft

Reply:
We do this in a lot of our higher level conveniences.  LRO polling only returns the last response.  Parallel upload for Blobs only returns the response of calling Commit.  Customers who care about every request can use lower level methods (the protocol method in this case) or submit batches small enough to fit in a single request.

Reply:
I think for LROs that behavior probably makes more sense as you are doing a bunch of requests serially and the last response is meaningful. Here, the response you get back is arbitrary and just represents whichever response came back last from a bunch of parallel requests. I wonder if we need a pattern for this kind of higher level helper that does not directly correspond to a particular response.

Reply:
For Blobs, there is a natural response to return - the response from Commit. There is no equivalent response here.

Reply:
I'm not really bothered about whether the last response of a batch is somehow less "natural."  If we ever added a parallel append helper, we wouldn't have a final commit and just return the last response that completed.  Our rule of thumb for conveniences is it must be _possible_ to see every response for customers who care a lot, but the happy path doesn't need to surface them all - especially in cases like this where there's nothing interesting about them.

Reply:
Agree that we don't need to surface all responses - but my thought was that it might be better to just not surface any responses here. Just throwing it out there as an observation.

Reply:
It might be good to discuss this across languages to make sure the behavior is consistent.